src/services/GISService/index.ts

```diff
@@ -0,0 +1,246 @@
+          }).filter(s => s !== '')
+
+          if (props.length === 0)
+          {
+            continue;
+          }
+
+          tmp.push(
+            `${ key }=${ conds.join(strict ? ' and ' : ' or ') }`
+          )
+        }
+      }
+      return '?' + tmp.join('&');
+    }
+
+    addslashes(str: string): string {
+      return str.replace(/\\/g, '\\\\').
+        replace(/\t/g, '\\t').
+        replace(/\n/g, '\\n').
+        replace(/\f/g, '\\f').
+        replace(/\r/g, '\\r').
+        replace(/'/g, '\\\'').
+        replace(/"/g, '\\"')
+    }
+  }
+
+  constructor () {
+    super(URL_BASE)
+  }
+
+  getInfoByZipCode (zipCode: number | string) {
+    return new Promise<GISResult>((resolve, reject) => {
+      const params = new GISSerice.ParamsBuilder()
+      params.setCQLFilter({ CITY: zipCode })
+
+      this.axios.get<GISResult>(params.build())
+        .then(({ data }) => {
+          data.features = data.features.map(f => {
+            if (typeof f.properties.INTERVENANTS === 'string') {
+              f.properties.INTERVENANTS = f.properties.INTERVENANTS.split(',')
+            }
+            return f
+          })
+          resolve(data as GISResult)
+        })
+        .catch((e: unknown) => {
+          reject(e)
+        })
+    })
+  }
+
+  getInfoByFilters (filters: IrismonumentProperties, strict = true) {
+    return new Promise<GISResult>((resolve, reject) => {
+      const params = new GISSerice.ParamsBuilder()
+      params.setCQLFilter(filters)
+      // console.log(params.build(strict))
+      this.axios.get<GISResult>(params.build(strict))
+        .then(({ data }) => {
+          data.features = data.features.map(f => {
+            if (typeof f.properties.INTERVENANTS === 'string') {
+              f.properties.INTERVENANTS = f.properties.INTERVENANTS.split(',')
+            }
+            return f
+          })
+          resolve(data as GISResult)
+        })
+        .catch((e: unknown) => reject(e))
+    })
+  }
+
+  getStats () {
+    return new Promise<Statistics>((resolve, reject) => {
+      const params = new GISSerice.ParamsBuilder()
+      console.log(params.build())
+      this.axios.get<GISResult>(params.build())
+        .then(({ data }) => {
+          const buildingsCount = data.totalFeatures
+
+          // number of buildings per zip code
+          const statsZIPCode = {} as {[key: string]: number}
+          data.features.forEach(f => {
+            if (!f.properties.CITY) {
+              return;
+            }
+
+            // if the zipCode already exists, increment its count, otherwise initialise it and set it's count to 1
+            if (statsZIPCode[f.properties.CITY]) {
+              statsZIPCode[f.properties.CITY]++
+            } else {
+              statsZIPCode[f.properties.CITY] = 1
+            }
+          })
+
+          // number of architectural styles per zip code
+          const municipalityStatistics = {} as {[key: string]: MunicipalityStatistics}
```

Comment:
Don't mix "municipality" and "postal code" !
If the statistics are based on the postal code (property `CITY`), please use "postal code" (instead of "municipality").